docs/book/how-to/customize-docker-builds/use-a-prebuilt-image.md

```diff
@@ -0,0 +1,120 @@
+{% endhint %}
+
+The following points are derived from how ZenML builds an image internally and will help you make your own images.
+
+### Your stack requirements
+
+A ZenML Stack can have different components and each comes with its own requirements. You need to ensure that your image contains them. The following is how you can get a list of stack requirements.
+
+```python
+from zenml.client import Client
+
+stack_name = <YOUR_STACK>
+# set your stack as active if it isn't already
+Client().set_active_stack(stack_name)
+
+# get the requirements for the active stack
+active_stack = Client().active_stack
+stack_requirements = active_stack.requirements()
+```
+
+### Integration requirements
+
+For all integrations that you use in your pipeline, you need to have their dependencies installed too. You can get a list of them in the following way:
+
+```python
+from zenml.integrations.registry import integration_registry
+from zenml.integrations.constants import HUGGINGFACE, PYTORCH
+
+# define a list of all required integrations
+required_integrations = [PYTORCH, HUGGINGFACE]
+
+# Generate requirements for all required integrations
+integration_requirements = set(
+    itertools.chain.from_iterable(
+        integration_registry.select_integration_requirements(
+            integration_name=integration,
+            target_os=OperatingSystemType.LINUX,
+        )
+        for integration in required_integrations
+    )
+)
+```
+
+### Any project-specific requirements
+
+For any other dependencies that your project relies on, you can then install all of these different requirements through a line in your `Dockerfile` that looks like the following. It assumes you have accumulated all the requirements in one file.
+
+```Dockerfile
+RUN pip install <ANY_ARGS> -r FILE
+```
+
+### Any system packages
+
+If you have any `apt` packages that are needed for your application to function, be sure to include them too. This can be achieved in a `Dockerfile` as follows:
+
+```Dockerfile
+RUN apt-get update && apt-get install -y --no-install-recommends YOUR_APT_PACKAGES
+```
+
+### Your project code files
+
+The files containing your pipeline and step code and all other necessary functions should also be available inside the image. 
+
+- If you have a code repository registered, you don't need to include your code files in the image yourself. ZenML will download them from the repository to the appropriate location in the image.
+
+- If you don't have a code repository but `allow_download_from_artifact_store` is set to `True` in your `DockerSettings` (`True` by default), ZenML will upload your code to the artifact store and make it available to the image.
+
+- If both of these options are disabled, you can include your code files in the image yourself.
+
+Take a look at [which files are built into the image](../../../../docs/book/how-to/customize-docker-builds/which-files-are-built-into-the-image.md) page to learn more about what to include.
```

Comment:
I think this whole paragraph should rather say that OPTIONALLY project code files should be there... tbh if you think about this, if you put the entire code, and environment and then try to run wth zenml, the whole use-case doesn't make sense... 

For me you should not be putting the code inside the docker image if you are using a pre-built docker image.. not sure what @schustmi thinkks

Reply:
I changed the wording a bit to reflect that the
- code should be available in the execution environment (not implying in the image itself)
- including code in the image (the third step) is not recommended.

open to what michael says